specs/system_config.md

```diff
@@ -0,0 +1,68 @@
+# System Config
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [System config contents](#system-config-contents)
+  - [`batcherHash` (`bytes32`)](#batcherhash-bytes32)
+  - [`l1FeeOverhead` and `l1FeeScalar` (`uint256,uint256`)](#l1feeoverhead-and-l1feescalar-uint256uint256)
+- [Writing the system config](#writing-the-system-config)
+- [Reading the system config](#reading-the-system-config)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+The `SystemConfig` is a contract on L1 that can emit rollup configuration changes as log events.
+The rollup [block derivation process](./derivation.md) picks up on these log events, and applies the changes.
+
+## System config contents
+
+### `batcherHash` (`bytes32`)
```

Comment:
Got it this makes sense, although we have multiple levels of versioning going on, since there is a top level config version and now the `batcherHash` itself is versioned and the first byte specifies the version.

Does this versioned batcher scheme only apply for system config version 0? That is where I think there is ambiguity here

Reply:
I think it makes sense to make each of the defined values properties of system config version 0 and then we can define system config version 1 later and if we need to use batcher hash version 1 before that then thats fine, and we can decide the semantics around the batcher hash with system config version 1 when it comes time

Reply:
> Does this versioned batcher scheme only apply for system config version 0?

Maybe, that's not been discussed yet. In the design doc phase we agreed on formatting it as bytes32. If you believe the event versioning is sufficient we can revert it back to just a bytes20 after discussing with design doc reviewers. I'm fine with it being either type.